packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -711,78 +785,92 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
+                            url,
+                            {
+                                body: file,
+                                headers,
+                                method: "POST",
+                            },
+                            "createBlob",
+                    ));
+                    event.end({
+                        blobId: res.content.id,
+                        ...res.propsToLog,
+                    });
+                    return res;
+                },
+            );
+        });
+
+        return response.content;
     }
+}
 
-    private async readTree(id: string): Promise<api.ISnapshotTree | null> {
-        if (!this.snapshotUrl) {
-            return null;
-        }
-        let tree = this.commitCache.get(id);
-        if (!tree) {
-            tree = await getWithRetryForTokenRefresh(async (options) => {
-                const storageToken = await this.getStorageToken(options, "ReadCommit");
-                const snapshotDownloader = async (url: string, fetchOptions: { [index: string]: any; }) => {
-                    return this.epochTracker.fetchAndParseAsJSON(
-                        url,
-                        fetchOptions,
-                        "snapshotTree",
-                    );
-                };
-                const snapshot = await fetchSnapshot(
-                    this.snapshotUrl!,
-                    storageToken,
-                    id,
-                    this.fetchFullSnapshot,
-                    !!this.hostPolicy.sessionOptions?.forceAccessTokenViaAuthorizationHeader,
-                    this.logger,
-                    snapshotDownloader,
-                );
-                let treeId = "";
-                if (snapshot.snapshotTree) {
-                    assert(snapshot.snapshotTree.id !== undefined, 0x222 /* "Root tree should contain the id!!" */);
-                    treeId = snapshot.snapshotTree.id;
-                    this.setRootTree(treeId, snapshot.snapshotTree);
-                }
-                if (snapshot.blobs) {
-                    this.initBlobsCache(snapshot.blobs);
-                }
-                // If the version id doesn't match with the id of the tree, then use the id of first tree which in that case
-                // will be the actual id of tree to be fetched.
-                return this.commitCache.get(id) ?? this.commitCache.get(treeId);
-            });
-        }
+export class LocalOdspDocumentStorageService extends OdspDocumentStorageServiceBase {
+    constructor(
+        resolvedUrl: IOdspResolvedUrl,
+        _logger: ITelemetryLogger,
+        private readonly fluidFile: Uint8Array | string,
+    ) {
+        super(
+            resolvedUrl,
+        );
+    }
 
-        if (!tree) {
-            return null;
+    public async getVersions(blobid: string | null, count: number): Promise<api.IVersion[]> {
+        let snapshotContents: ISnapshotContents;
```

Comment:
I think we should assert here that count === 1 && blobid === null. Anything else seems like a suspect.

Reply:
Also, is there any harm if this call is done twice?